pennylane/operation.py

```diff
@@ -964,6 +964,77 @@ def kraus_matrices(self):
         return self._kraus_matrices(*self.parameters)
 
 
+class Projection(Operation):
+    """Base class for projection operations
+
+    As with :class:`~.Operation`, the following class attributes must be
+    defined for all projections:
+
+    * :attr:`~.Operator.num_params`
+    * :attr:`~.Operator.num_wires`
+    * :attr:`~.Operator.par_domain`
+
+    To define a projection, the following attribute of :class:`~.Projection`
+    can be used to list the corresponding projectors.
+
+    * :attr:`~.Projection._projectors`
+
+    Args:
+        params (tuple[float, int, array, Variable]): operation parameters
+
+    Keyword Args:
+        wires (Sequence[int]): Subsystems the projection acts on. If not given, args[-1]
+            is interpreted as wires.
+        do_queue (bool): Indicates whether the operation should be
+            immediately pushed into a :class:`BaseQNode` circuit queue.
+            This flag is useful if there is some reason to run an Operation
+            outside of a BaseQNode context.
+    """
+
+    # pylint: disable=abstract-method
```

Comment:
Do we need this, since we don't have any abstract methods?
On the other hand _should_ we have abstract methods?
`Channel` does declare `_kraus_matrices` as an abstract method, but `Operator` doesn't have `_matrix` as an abstract method, so we're not consistent :thinking: 

Maybe it's fine here to keep as is (just remove the pylint ignore - unless I missed something!)

Reply:
Right, maybe we should remove it. Codecov is throwing a fit over us not testing it :joy:

Reply:
> Channel does declare _kraus_matrices as an abstract method, but Operator doesn't have _matrix as an abstract method, so we're not consistent 🤔

Ah 🤔 

I think this is an oversight, needing to define `_kraus_matrices` shouldn't be an abstract method.